src/core/time.jl

```diff
@@ -1,38 +1,66 @@
+#  General TODO:  
+#  1. Confirm that we can use the general case for setindex!, getindex,
+# 	and has value instead of teh two specific cases for Variable and Fixed.  
+#	Is there any reason to specify Start, Step or Year here in order to be
+#	positive that they match?
+
 #
-#  TIMESTEP
+#  1. TIMESTEP
 #
-function gettime{Start, Step, Stop}(ts::Timestep{Start, Step, Stop})
+
+function gettime(ts::Timestep{Start, Step, Stop}) where {Start, Step, Stop}
 	return Start + (ts.t - 1) * Step
 end
 
-function is_start(ts::Timestep)
+function gettime(ts::VariableTimestep)
+	return ts.current
+end
+
+function is_start(ts::AbstractTimestep)
 	return ts.t == 1
 end
 
-# indicates when at final timestep
-function is_stop{Start, Step, Stop}(ts::Timestep{Start, Step, Stop})
+# LFR-TBD:  is_stop function is not used internally, so we may want to deprecate it ... 
+# look into where it might be used within models?
+function is_stop(ts::Timestep{Start, Step, Stop}) where {Start, Step, Stop}
 	return gettime(ts) == Stop
 end
 
-# used to determine when a clock is finished
-function finished{Start, Step, Stop}(ts::Timestep{Start, Step, Stop})
+function is_stop(ts::VariableTimestep{Years}) where {Years}
+	return gettime(ts) == Years[end]
+end
+
+function finished(ts::Timestep{Start, Step, Stop}) where {Start, Step, Stop}
 	return gettime(ts) > Stop
 end
 
-function next_timestep{Start, Step, Stop}(ts::Timestep{Start, Step, Stop})
+function finished(ts::VariableTimestep{Years}) where {Years}
```

Comment:
Looks like these two funcs can be combined, since the type parameters of the two Timestep variants are not used at all.

```
function next_timestep(ts::Timestep{Start, Step, Stop}) where {Start, Step, Stop}
	if finished(ts)
			error("Cannot get next timestep, this is final timestep.")
	end
	return Timestep{Start, Step, Stop}(ts.t + 1)
end

function next_timestep(ts::VariableTimestep{start_times}) where {start_times}
	if finished(ts)
		error("Cannot get next timestep, this is final timestep.")
	end
	return VariableTimestep{start_times}(ts.t + 1)		
end
```
into
```
function next_timestep(ts::T) where {T :< AbstractTimestep}
	if finished(ts)
	    error("Cannot get next timestep, this is final timestep.")
	end
	return T(ts.t + 1)
end
```